bgpd/bgp_mplsvpn.c

```diff
@@ -989,122 +987,135 @@ DEFUN (show_bgp_ipv4_vpn_rd,
+      if (! ret)
+        {
+          vty_out (vty, "%% Malformed Route Distinguisher%s", VTY_NEWLINE);
+          return CMD_WARNING;
+        }
+      return bgp_show_mpls_vpn (vty, afi, &prd, bgp_show_type_normal, NULL, 0, 0);
     }
-  return bgp_show_mpls_vpn (vty, AFI_IP, &prd, bgp_show_type_normal, NULL, 0, 0);
-}
+  return CMD_SUCCESS;
+ }
 
-DEFUN (show_ip_bgp_vpnv4_all_tags,
-       show_ip_bgp_vpnv4_all_tags_cmd,
-       "show [ip] bgp vpnv4 all tags",
+DEFUN (show_ip_bgp_vpn_all_tags,
+       show_ip_bgp_vpn_all_tags_cmd,
+       "show [ip] bgp <vpnv4|vpnv6> all tags",
        SHOW_STR
        IP_STR
        BGP_STR
        "Address Family\n"
-       "Display information about all VPNv4 NLRIs\n"
+       BGP_VPNVX_HELP_STR
+       "Display information about all VPNv4/VPNV6 NLRIs\n"
        "Display BGP tags for prefixes\n")
 {
-  return bgp_show_mpls_vpn (vty, AFI_IP, NULL, bgp_show_type_normal, NULL,  1, 0);
+  afi_t afi;
+  int idx = 0;
+
+  if (argv_find_and_parse_vpnvx (argv, argc, &idx, &afi))
+    return bgp_show_mpls_vpn (vty, afi, NULL, bgp_show_type_normal, NULL,  1, 0);
+  return CMD_SUCCESS;
 }
 
-DEFUN (show_ip_bgp_vpnv4_rd_tags,
-       show_ip_bgp_vpnv4_rd_tags_cmd,
-       "show [ip] bgp vpnv4 rd ASN:nn_or_IP-address:nn tags",
+DEFUN (show_ip_bgp_vpn_rd_tags,
+       show_ip_bgp_vpn_rd_tags_cmd,
+       "show [ip] bgp <vpnv4|vpnv6> rd ASN:nn_or_IP-address:nn tags",
        SHOW_STR
        IP_STR
        BGP_STR
        "Address Family\n"
+       BGP_VPNVX_HELP_STR
        "Display information for a route distinguisher\n"
        "VPN Route Distinguisher\n"
        "Display BGP tags for prefixes\n")
 {
   int idx_ext_community = 5;
   int ret;
   struct prefix_rd prd;
+  afi_t afi;
+  int idx = 0;
 
-  ret = str2prefix_rd (argv[idx_ext_community]->arg, &prd);
-  if (! ret)
+  if (argv_find_and_parse_vpnvx (argv, argc, &idx, &afi))
     {
-      vty_out (vty, "%% Malformed Route Distinguisher%s", VTY_NEWLINE);
-      return CMD_WARNING;
+      ret = str2prefix_rd (argv[idx_ext_community]->arg, &prd);
+      if (! ret)
+        {
+          vty_out (vty, "%% Malformed Route Distinguisher%s", VTY_NEWLINE);
+          return CMD_WARNING;
+        }
+      return bgp_show_mpls_vpn (vty, afi, &prd, bgp_show_type_normal, NULL, 1, 0);
     }
-  return bgp_show_mpls_vpn (vty, AFI_IP, &prd, bgp_show_type_normal, NULL, 1, 0);
+  return CMD_SUCCESS;
 }
 
-DEFUN (show_ip_bgp_vpnv4_all_neighbor_routes,
-       show_ip_bgp_vpnv4_all_neighbor_routes_cmd,
-       "show [ip] bgp vpnv4 all neighbors A.B.C.D routes [json]",
+DEFUN (show_ip_bgp_vpn_all_neighbor_routes,
+       show_ip_bgp_vpn_all_neighbor_routes_cmd,
+       "show [ip] bgp <vpnv4|vpnv6> all neighbors A.B.C.D routes [json]",
        SHOW_STR
        IP_STR
        BGP_STR
        "Address Family\n"
-       "Display information about all VPNv4 NLRIs\n"
+       BGP_VPNVX_HELP_STR
+       "Display information about all VPNv4/VPNv6 NLRIs\n"
```

Comment:
about stable/2.0 and master, I am a bit confused. 
Since I do some work effort to porting some features, I want to avoid being in the wrong branch.
For instance, I rebase evpn work with master branch.